### DISCUSSION TITLE
The decorator pattern using .NET's dependency injection

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#369] [Announcements]
# The decorator pattern using .NET's dependency injection

Taking leverage of the decorator pattern to add extra behavior to existing an implementation. We first see how to do this manually, before taking a look at Scrutor to simplify this task.

https://timdeschryver.dev/blog/the-decorator-pattern-using-nets-dependency-injection

**Comment by btull89** [comment]
Thank you so much for this article. This was extremely helpful when trying to implement a decorator myself!

**Reply by timdeschryver** [reply]
I'm glad to hear that @btull89 !
